### DISCUSSION TITLE
How to get input in NASM X86?

### DISCUSSION THREAD

**Discussion by emullallphotography** [community/community#72256] [Programming Help]
### Body

Hi,

I am coding a text based version of Snake in NASM X86 and I got the input but know I need to position the characters to random places in the terminal, get user input and code the collision detection

### Guidelines

- [x] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by RodEspinoza** [comment]
Hi . @emullallphotography 
i was researching, this probably could help :

```
the_game_loop:
    ; Your main game logic happens here

    ; Read user input
    mov eax, 0
    mov ebx, 0
    mov ecx, input
    mov edx, 4
    int 0x80

    ; Check if the input is an arrow key //
    cmp byte [input], 27
    jne no_arrow

 ; The input corresponds to an arrow key escape sequence
    cmp byte [input + 2], 'D'  ; Left arrow key
    je left_arrow

    cmp byte [input + 2], 'C'  ; Right arrow key
    je right_arrow

no_arrow:
    ; Other actions as ignore the key 

    ; Jump to the end of the loop to continue reading user input
    jmp end_loop

left_arrow:
    ; Move the cursor to the left or perform actions related to the left arrow
    ; You can use system calls to control the cursor position on the terminal

    ; Translate and display a message or move your snake head 
    mov eax, 4
    mov ebx, 1
    mov ecx, left_message
    mov edx, 8
    int 0x80

 jmp end_loop

```

**Comment by LiroSiza** [comment]
I created a project for NASM x86 of the snake game through the terminal. I used a C program to paint the terminal. To capture user input, I switched the terminal to raw mode to detect keys without waiting for an enter key. Regarding collisions, I detected them using a vector that represented the snake.
You can check the repository for more information; I hope it is helpful to you.
https://github.com/RobFillingham/Snake
